eng/scripts/Invoke-GenerateAndBuildV2.ps1

```diff
@@ -108,6 +109,39 @@ if ($inputFileToGen) {
     UpdateExistingSDKByInputFiles -inputFilePaths $inputFileToGen -sdkRootPath $sdkPath -headSha $commitid -repoHttpsUrl $repoHttpsUrl -downloadUrlPrefix "$downloadUrlPrefix" -generatedSDKPackages $generatedSDKPackages
 }
 
+# generate sdk from cadl file
+if ($relatedCadlProjectFolder) {
+    for ($i = 0; $i -le $relatedCadlProjectFolder.Count - 1; $i++) {
+        $caldFolder = (Join-Path $swaggerDir $relatedCadlProjectFolder[$i]) -replace "\\", "/"
+        $newPackageOutput = "newPackageOutput.json"
+
+        Push-Location $caldFolder
+        $cadlProjectYaml = Get-Content -Path "$caldFolder/cadl-project.yaml" -Raw
+
+        Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+        $yml = ConvertFrom-YAML $cadlProjectYaml
+        $sdkFolder = $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"]
+        $projectFolder = (Join-Path $sdkPath $sdkFolder)
+        $projectFolder = $projectFolder -replace "\\", "/"
+        if ($projectFolder) {
+            $directories = $projectFolder.Split("/");
+            $count = $directories.Count
+            $projectFolder = $directories[0 .. ($count-2)] -join "/"
+            $service = $directories[-3];
+            $namespace = $directories[-2];
+        }
+        New-CADLPackageFolder -service $service -namespace $namespace -sdkPath $sdkPath -cadlInput $caldFolder/main.cadl -outputJsonFile $newpackageoutput
+        $newPackageOutputJson = Get-Content $newPackageOutput | Out-String | ConvertFrom-Json
```

Comment:
The `out-string` is unnecessary. The content is already a string and `convertfrom-json` handles both a string and string array correctly (unless you're on *really* older Windows PowerShell, which you shouldn't be using...it's effectively defunct.

Reply:
resolved